### PR TITLE
TSWEB-670 Add tests for UXUI design

### PR DESCRIPTION
**Pull request by AnnaSlisenko**

### CODE REVIEW

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+		'Mobile\ndevelopment',
+		'Consulting',
+		'Custom software\ndevelopment',
+		'AI & ML',
+		'Big Data &\nAnalytics',
+		'IoT',
+	]);
+});
+
+test("Check Faq container from the 'UX/UI Design' block @Regression @UxUiDesign @TSWEB-670", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts:267]: What do you certainly check in this test? What's mean 'Check Faq container'? Try to give more concrete info about certain test-case and what it checks.

**Review by VladFedash — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/identifiers/UxUiDesign.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	static GetCustomUxAndUiDesignServices = 'Container-GetCustomUxAndUiDesignServices';
+	static WeBuildUxUiForMobileWeb = 'Container-WeBuildUxUiForMobileWeb';
+	static InDesignWeTrust = 'Container-InDesignWeTrust';
+	static CaseStudies = 'Container-CaseStudies';
+	static TypicalUxUiDesignWorkflow = 'Container-TypicalUxUiDesignWorkflow';
+	static WeNeverStopImprovingYourProduct = 'Container-WeNeverStopImprovingYourProduct';
+	static RelatedServices = 'Container-RelatedServices';
+	static Faq = 'Container-Faq';
+
+	//Button
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/identifiers/UxUiDesign.ts:12]: I`ll fix it

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+	);
+	await expect(driver.getByTestId(UxUiDesign.RelatedServices).getByTestId(Container.ContainerNumber)).toHaveText(
+		'07'
+	);
+
+	await expect(driver.getByTestId(UxUiDesign.Faq).getByTestId(Container.ContainerTitle)).toHaveText('FAQ');
+	await expect(driver.getByTestId(UxUiDesign.Faq).getByTestId(Container.ContainerNumber)).toHaveText('08');
+});
+
+test("Check GetCustomUxAndUiDesignServices container from the 'UX/UI Design' block @Regression @UxUiDesign @TSWEB-670", async () => {
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts:71]: Ok, I`ll change namings

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+		'Elizabeth Malygina',
+		'Hanna Zhyhan',
+	]);
+});
+
+test("Check links in WeNeverStopImprovingYourProduct container from the 'UX/UI Design' block @Regression @UxUiDesign @TSWEB-670", async () => {
+	const weNeverStopImprovingYourProduct = driver.getByTestId(UxUiDesign.WeNeverStopImprovingYourProduct);
+
+	const linkMap = new Map([
+		[UxUiDesign.Instagram, 'https://www.instagram.com/techstack.design/'],
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts:219]: I suppose it is not necessary. These links are used only in current container

**Review by VladFedash — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+		const [newPage] = await Promise.all([
+			driver.DriverContext.waitForEvent('page'),
+			await weNeverStopImprovingYourProduct.getByTestId(entries[0]).first().click(),
+		]);
+		expect(newPage.url()).toContain(entries[1]);
+		await newPage.close();
+	}
+});
+
+test("Check RelatedServices container from the 'UX/UI Design' block @Regression @UxUiDesign @TSWEB-670", async () => {
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts]: This line is needed to continue the "for"  loop which checks correct redirections via arrows
It means like an "after" step, not "when" meaning)

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+	const linkMap = new Map([
+		[UxUiDesign.Instagram, 'https://www.instagram.com/techstack.design/'],
+		[UxUiDesign.Tiktok, 'https://www.tiktok.com/@techstack.design'],
+		[UxUiDesign.Behance, 'https://www.behance.net/dimadityuk'],
+		[Container.MemberCard, 'https://www.behance.net/dimadityuk'],
+		[Container.MemberName, 'https://www.behance.net/dimadityuk'],
+	]);
+
+	for (const entries of linkMap.entries()) {
+		const [newPage] = await Promise.all([
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts]: Fixed

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+	const arrowList = await relatedServices.getByTestId(Container.Arrow).all();
+
+	for (let index = 0; index < arrowUrlList.length; index++) {
+		await arrowList[index].click();
+		await baseDriverSteps.checkUrl(arrowUrlList[index]);
+		await baseDriverSteps.goToUrl(UrlProvider.urlBuilder(UrlPath.UiUxDesign));
+	}
+});
+
+test("Check redirects in RelatedServices container from the 'UX/UI Design' block @Regression @UxUiDesign @TSWEB-670", async () => {
```
[the remark this answers is not in the document]
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts]: Fixed

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts (hunk 1 above)
Comment (AnnaSlisenko) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts]: Fixed

**Review by anastasiia-liashova — APPROVED** [review]

File: ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts
[the remark this answers is not in the document]
Comment (VladFedash) [reply ui_tests_playwright/automation/test/ui/services/UxUiDesign.test.ts:1]: I anyway think it's actual comment

**Review by VladFedash — APPROVED** [review]

**Review by DaniilVostrikov — CHANGES_REQUESTED** [review]

**Review by anastasiia-liashova — APPROVED** [review]

**Review by DaniilVostrikov — APPROVED** [review]

**Pull request merged**